kythe/typescript/indexer.ts

```diff
@@ -308,30 +303,36 @@ class Visitor {
         {...this.kFile}, {signature: `@${start}:${end}`, language: LANGUAGE});
     this.emitNode(name, 'anchor');
     const offsetTable = this.getOffsetTable(node.getSourceFile().fileName);
-    this.emitFact(name, 'loc/start', offsetTable.lookup(start).toString());
-    this.emitFact(name, 'loc/end', offsetTable.lookup(end).toString());
+    this.emitFact(
+        name, FactName.LOC_START, offsetTable.lookup(start).toString());
+    this.emitFact(name, FactName.LOC_END, offsetTable.lookup(end).toString());
     return name;
   }
 
   /** emitNode emits a new node entry, declaring the kind of a VName. */
   emitNode(source: VName, kind: string) {
-    this.emitFact(source, 'node/kind', kind);
+    this.emitFact(source, FactName.NODE_KIND, kind);
+  }
+
+  /** emitSubkind emits a new fact entry, declaring the subkind of a VName. */
+  emitSubkind(source: VName, subkind: Subkind) {
+    this.emitFact(source, FactName.SUBKIND, subkind);
   }
 
   /** emitFact emits a new fact entry, tying an attribute to a VName. */
-  emitFact(source: VName, name: string, value: string) {
+  emitFact(source: VName, name: FactName, value: string) {
     this.host.emit({
       source,
-      fact_name: '/kythe/' + name,
+      fact_name: name,
       fact_value: Buffer.from(value).toString('base64'),
     });
   }
 
   /** emitEdge emits a new edge entry, relating two VNames. */
-  emitEdge(source: VName, name: string, target: VName) {
+  emitEdge(source: VName, name: EdgeKind|OrdinalEdge, target: VName) {
     this.host.emit({
       source,
-      edge_kind: '/kythe/edge/' + name,
+      edge_kind: name,
```

Comment:
This is weird, should the parameter be 'kind'?

Reply:
yes